internal/controller/pdbwatcher_controller.go

```diff
@@ -161,7 +206,35 @@ func (r *PDBWatcherReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		}
 	}
 
-	// What will return us to the original state? Assuming we need a watch on PDB for it to come off
+	// Watch for changes in PDB to revert to original state
+	if pdb.Status.DisruptionsAllowed > 0 && *deployment.Spec.Replicas != pdbWatcher.Status.MinReplicas {
+		// Check if the resource version has changed
+		err = r.Get(ctx, types.NamespacedName{Name: deploymentName, Namespace: pdbWatcher.Namespace}, deployment)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		if deployment.ResourceVersion != initialResourceVersion {
+			// Deployment has been modified externally, revert to the original state
```

Comment:
at this point you haven't changed it yet so you don't need to touch deployment just reset your min replicas.